bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -349,17 +374,27 @@ private class SessionListenerAdapter extends Session.Listener.Adapter {
 
         @Override
         public void onClose(@Nullable Session session, @Nullable GoAwayFrame frame) {
-            fatalErrorDelayed(this, new Http2Exception(Http2Error.CLOSED));
+            Objects.requireNonNull(session);
+            fatalErrorDelayed(this, new Http2Exception(Http2Error.CLOSED), session.hashCode(), -1);
         }
 
         @Override
         public void onFailure(@Nullable Session session, @Nullable Throwable failure) {
-            fatalErrorDelayed(this, new Http2Exception(Http2Error.FAILURE));
+            Objects.requireNonNull(session);
+            fatalErrorDelayed(this, new Http2Exception(Http2Error.FAILURE), session.hashCode(), -1);
         }
 
+        /**
+         * The Hue bridge uses the 'nginx' web server which sends HTTP2 GO_AWAY frames after a certain number (normally
+         * 999) of GET/PUT calls. This is normal behaviour so we just start a new thread to close and reopen (recycle)
+         * the session.
+         */
         @Override
         public void onGoAway(@Nullable Session session, @Nullable GoAwayFrame frame) {
-            fatalErrorDelayed(this, new Http2Exception(Http2Error.GO_AWAY));
+            Objects.requireNonNull(session);
+            if (http2Session == session) {
+                new Thread(() -> recycleSession()).start();
```

Comment:
Why not use the scheduler here?

Reply:
Because it has to be REALLY fast, and without risk of any external delay/dependency.

Reply:
Curious, what would happen if it's not really fast?